app/code/Magento/Quote/Model/ResourceModel/Quote.php

```diff
@@ -316,4 +316,28 @@ public function save(\Magento\Framework\Model\AbstractModel $object)
 
         return $this;
     }
+
+    /**
+     * Quickly check if quote exists
+     *
+     * Uses direct DB query due to performance reasons
+     *
+     * @param int $quoteId
+     * @return bool
+     */
+    public function isQuoteExists(int $quoteId): bool
```

Comment:
I believe `exists` as method name is more suitable in this case. Since we are already in `Quote` ResourceModel, there is no reason to double that info in method name as well.
```suggestion
    public function exists(int $quoteId): bool
```

Reply:
Maybe it's better to add "is" prefix?
```suggestion
    public function isExists(int $quoteId): bool
```

Reply:
I don't think it's a good idea to have names with 2 verbs.
- isVirtual
- isBlocked
So, I'd stick to `exists` method name.